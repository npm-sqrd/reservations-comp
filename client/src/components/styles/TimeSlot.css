:local(.container) {
  /*border: 1px solid purple;*/
}

:local(.button) {
  font-family: BrandonText,-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  background-color: #da3743;
  color: #FFF;
  width: 120px;
  height: 35px;
  border-radius: 2px;
  margin: 5px;
  font-size: 15px;
}

:local(.button:hover) {
  background-color: #b22c36;
  cursor: pointer;
}


:local(.button:disabled) {
  background-color: #AAA;
  cursor: default;
}
