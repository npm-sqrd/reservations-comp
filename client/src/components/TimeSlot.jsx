import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles/TimeSlot.css';

const TimeSlot = props => (
  <div className={styles.container}>
    <button
      className={styles.button}
      disabled={!props.available}
      onClick={() => props.clickHandler(props.time)}
    >
      {props.time > 12 ? props.time - 12 : props.time}:00 {props.time > 12 ? 'PM' : 'AM'}
    </button>
  </div>
);


TimeSlot.propTypes = {
  time: PropTypes.number.isRequired,
  available: PropTypes.bool.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default TimeSlot;
