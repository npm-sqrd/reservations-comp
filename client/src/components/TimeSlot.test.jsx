import React from 'react';
import { shallow } from 'enzyme';

import TimeSlot from './TimeSlot';

describe('TimeSlot Component', () => {
  test('should render available PM times correctly', () => {
    expect(shallow(<TimeSlot
      time={17}
      available
      clickHandler={() => {}}
    />)).toMatchSnapshot();
  });

  test('should render available AM times correctly', () => {
    expect(shallow(<TimeSlot
      time={10}
      available
      clickHandler={() => {}}
    />)).toMatchSnapshot();
  });


  test('should render unavailable times correctly', () => {
    expect(shallow(<TimeSlot
      time={17}
      available={false}
      clickHandler={() => {}}
    />)).toMatchSnapshot();
  });

  test('should be disabled if not available', () => {
    const mockHandler = jest.fn();
    const component = shallow(<TimeSlot
      time={17}
      available={false}
      clickHandler={mockHandler}
    />);

    const button = component.find('button').first();

    expect(button.props().disabled).toBe(true);
  });

  test('should call clickHandler function with appropriate arguments when clicked', () => {
    const mockHandler = jest.fn();
    const component = shallow(<TimeSlot
      time={17}
      available
      clickHandler={mockHandler}
    />);

    const button = component.find('button').first();
    button.props().onClick();

    expect(mockHandler.mock.calls.length).toBe(1);
    expect(mockHandler.mock.calls[0][0]).toBe(17);
  });
});
