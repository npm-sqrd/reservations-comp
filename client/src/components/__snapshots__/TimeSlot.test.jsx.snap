// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlot Component should render available AM times correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeSlot
    available={true}
    clickHandler={[Function]}
    time={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className={undefined}
        disabled={false}
        onClick={[Function]}
      >
        10
        :00 
        AM
      </button>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          10,
          ":00 ",
          "AM",
        ],
        "className": undefined,
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        10,
        ":00 ",
        "AM",
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className={undefined}
          disabled={false}
          onClick={[Function]}
        >
          10
          :00 
          AM
        </button>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            10,
            ":00 ",
            "AM",
          ],
          "className": undefined,
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          10,
          ":00 ",
          "AM",
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TimeSlot Component should render available PM times correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeSlot
    available={true}
    clickHandler={[Function]}
    time={17}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className={undefined}
        disabled={false}
        onClick={[Function]}
      >
        5
        :00 
        PM
      </button>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          5,
          ":00 ",
          "PM",
        ],
        "className": undefined,
        "disabled": false,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        5,
        ":00 ",
        "PM",
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className={undefined}
          disabled={false}
          onClick={[Function]}
        >
          5
          :00 
          PM
        </button>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            5,
            ":00 ",
            "PM",
          ],
          "className": undefined,
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          5,
          ":00 ",
          "PM",
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TimeSlot Component should render unavailable times correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimeSlot
    available={false}
    clickHandler={[Function]}
    time={17}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className={undefined}
        disabled={true}
        onClick={[Function]}
      >
        5
        :00 
        PM
      </button>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          5,
          ":00 ",
          "PM",
        ],
        "className": undefined,
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        5,
        ":00 ",
        "PM",
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className={undefined}
          disabled={true}
          onClick={[Function]}
        >
          5
          :00 
          PM
        </button>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            5,
            ":00 ",
            "PM",
          ],
          "className": undefined,
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          5,
          ":00 ",
          "PM",
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
